crates/breez-sdk/core/src/sdk.rs

```diff
@@ -228,6 +229,21 @@ impl BreezSdk {
     fn start(&self, initial_synced_sender: watch::Sender<bool>) {
         self.periodic_sync(initial_synced_sender);
         self.try_recover_lightning_address();
+        self.apply_spark_private_mode_setting();
```

Comment:
That may fail and users would think they are in private mode.
Perhaps an event such as UpdateWalletSettingFailed?
Also should we expose update_wallet_settings function in case users want to retry?

Reply:
We may add an event but I fear the UX will be suboptimal. I don't like that users would be required to handle/look for the event to know something is wrong.

I'm looking into the possibility of adding automatic retries and preventing payments before the it eventually succeeds. Wdyt?

Reply:
Added retries and a check before payments. Also, I propose we only enforce the setting was applied in case it is on. If turned off, I don't think we need to prevent payments.